### DISCUSSION TITLE
Xcodebuild failed

### DISCUSSION THREAD

**Discussion by denandreychuk** [swiftyfinch/Rugby#397] [Q&A] [answered]
⛔️ Error: Xcodebuild failed.
 ✖ ld: framework 'WatchKit' not found
 ✖ clang: linker command failed with exit code 1 (use -v to see invocation)
 ✖ ld: warning: directory not found for option '-L.rugby/build/Debug-iphoneos/XCFrameworkIntermediates/Realm'
 ✖ ld: library not found for -lrealm-monorepo
 ✖ clang: linker command failed with exit code 1 (use -v to see invocation)
 ✖ ld: warning: directory not found for option '-L.rugby/build/Debug-iphoneos/XCFrameworkIntermediates/Realm'
 ✖ ld: library not found for -lrealm-monorepo
 ✖ clang: linker command failed with exit code 1 (use -v to see invocation)
 ✖ ld: framework not found Realm
 ✖ clang: linker command failed with exit code 1 (use -v to see invocation)
 ✖ ld: framework not found Realm
 ✖ clang: linker command failed with exit code 1 (use -v to see invocation)
 ✖ Multiple commands produce '.rugby/build/Debug-iphoneos/XCFrameworkIntermediates/Realm/realm-monorepo.framework'

**Comment by denandreychuk** [comment]
In case you need commands:

bundle exec pod install
rugby build --arch arm64 --sdk sim
rugby build --arch arm64 --sdk ios
rugby use

**Reply by swiftyfinch** [reply, marked as answer]
Hello @denandreychuk!

First of all, could you please provide an example to reproduce the issue?
This would be greatly appreciated.

Do you include targets with different options (such as `use_frameworks!`) in your Podfile? If so, you should only build one target at a time, as Rugby builds all targets in the Pods project by default.

For example, take a look at [the Podfile of the Rugby Example project](https://github.com/swiftyfinch/Rugby/blob/main/Example/Podfile). There are two different targets defined there:
```ruby
target 'ExampleFrameworks' do
  use_frameworks!
  ...
end

target 'ExampleLibs' do
  use_modular_headers!
end
```
They group similar pods together, but in a different way.
You can build the configuration like this:
```sh
> rugby build --targets Pods-ExampleFrameworks --arch arm64 --sdk sim
> rugby build --targets Pods-ExampleFrameworks --arch arm64 --sdk ios
> rugby use --targets Pods-ExampleFrameworks
```
Or like this:
```sh
> rugby build --targets Pods-ExampleLibs --arch arm64 --sdk sim
> rugby build --targets Pods-ExampleLibs --arch arm64 --sdk ios
> rugby use --targets Pods-ExampleLibs
```
